todo_app/app.py

```diff
@@ -1,11 +1,23 @@
-from flask import Flask
+from flask import Flask, render_template, request, jsonify
+from todo_app.data.session_items import get_items, add_item
 
 from todo_app.flask_config import Config
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
 
-@app.route('/')
+@app.route('/', methods = ['GET','POST'])
 def index():
-    return 'Hello World!'
+    if request.method == 'GET':
+        items = get_items()
+        return render_template('index.html', items = items)
+    
+    elif request.method == 'POST':
+        item = request.form['item']
+        add_item(item)
+        items = get_items()
+        return render_template('index.html', items = items)
+
+    else:
+        return jsonify({"message":"error"})
```

Comment:
Preference rather than strong recommendation, but I'm a fan of splitting the function here out to handle the two different methods separately, and we can then rely on Flask to ensure the `else` clause is never required:
```python
@app.route('/', methods = ['GET'])
def index():
    items = get_items()
    return render_template('index.html', items = items)

@app.route('/', methods = ['POST'])
def add_item():
    item = request.form['item']
    add_item(item)
    items = get_items()
    return render_template('index.html', items = items)
```

And then to take that one further, note that this means we'll need to duplicate any changes in the first function in the second, so rather than returning directly, we can redirect the user back to the main page:
```python
@app.route('/', methods = ['POST'])
def add_item():
    item = request.form['item']
    add_item(item)
    return redirect('/')
```
(relies on adding an import from Flask for `redirect`)